pithos/pithos.py

```diff
@@ -90,21 +106,96 @@ def do_set_property(self, pspec, value):
-            Gdk.cairo_set_source_pixbuf(ctx, pixbuf, x, y)
+            if self.icon == 'love':
+                rating_icon = self.love_icon
+            elif self.icon == 'tired':
+                rating_icon = self.tired_icon
+            elif self.icon == 'ban':
+                rating_icon = self.ban_icon
+
+            x = x + (rating_icon.get_width() // 2)
+            y = y + (rating_icon.get_height() // 2)
+
+            Gdk.cairo_set_source_pixbuf(ctx, rating_icon, x, y)
             ctx.paint()
 
+    def update_icons(self, style_context):
+        #Dynamically change the color of backgrounds and icons
+        #to match the current theme at theme changes.
+        #Attempt to look up the background and foreground colors
+        #in the theme's CSS file. Otherwise if they aren't found
+        #fallback to get_background_color(Deprecated since version 3.16)
+        #and get_color. *Most* new themes use 'theme_bg_color' and 'theme_fg_color'.
+        #Some(older) themes use 'bg_color' and 'fg_color'.(like Ubuntu light themes)
+        bg_bool, bg_color = style_context.lookup_color('theme_bg_color')
+        if not bg_bool:
+            bg_bool, bg_color = style_context.lookup_color('bg_color')
+            if not bg_bool:
+                bg_color = style_context.get_background_color(Gtk.StateFlags.NORMAL)
+                logging.debug("Could not find 'theme_bg_color' or 'bg_color' with lookup_color.")
+
+        fg_bool, fg_color = style_context.lookup_color('theme_fg_color')
+        if not fg_bool:
+            fg_bool, fg_color = style_context.lookup_color('fg_color')
+            if not fg_bool:
+                fg_color = style_context.get_color(Gtk.StateFlags.NORMAL)
+                logging.debug("Could not find 'theme_fg_color' or 'fg_color' with lookup_color.")
+
+        fg_rgb = fg_color.to_string()
+        bg_rgb = bg_color.to_string()
+
+        #Use our color values to create strings representing valid SVG's
+        #for backgound and rate_bg, then load them with PixbufLoader.
+        background = BACKGROUND_SVG.format(px=ALBUM_ART_SIZE, fg=fg_rgb).encode()
+        rating_bg = RATING_BG_SVG.format(bg=bg_rgb).encode()
+
+        with contextlib.closing(GdkPixbuf.PixbufLoader()) as loader:
+            loader.write(background)
+        self.background = loader.get_pixbuf()
+
+        with contextlib.closing(GdkPixbuf.PixbufLoader()) as loader:
+            loader.write(rating_bg)
+        self.rate_bg = loader.get_pixbuf()
+
+        current_theme = Gtk.IconTheme.get_default()
+
+        #Pithos requires an icon theme with symbolic icons.
+
+        #Manually color audio-x-generic-symbolic 48px icon to be used as part of the "default cover".
+        info = current_theme.lookup_icon('audio-x-generic-symbolic', 48, 0)
+        self.generic_audio_icon, was_symbolic = info.load_symbolic(bg_color, bg_color, bg_color, bg_color)
+
+        #We request 24px icons because what we really want is 12px icons,
```

Comment:
__[flake8]__
* [E265] block comment should start with '# '

<!-- Comment by [SideCI](https://sideci.com) -->